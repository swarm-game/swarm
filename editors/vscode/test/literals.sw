def lit = \v. log (format v) end

lit 15;

lit 0xf;

lit 0b1111;

lit " Hello ";

lit "Hello\nWorld!";
