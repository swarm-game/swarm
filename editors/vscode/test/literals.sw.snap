>def lit = \v. log (format v) end
#^^^^ source.swarm keyword.control.dictionary.def
#    ^^^ source.swarm keyword.control.dictionary.def entity.name.function
#       ^ source.swarm keyword.control.dictionary.def
#        ^ source.swarm keyword.control.dictionary.def keyword.control.dictionary.def.assignment
#         ^ source.swarm
#          ^ source.swarm keyword.operator.lambda
#           ^ source.swarm keyword.operator.lambda variable.other
#            ^ source.swarm keyword.operator.lambda
#             ^ source.swarm
#              ^^^ source.swarm keyword.other
#                 ^ source.swarm
#                  ^ source.swarm expression.group punctuation.paren.open
#                   ^^^^^^ source.swarm expression.group keyword.other
#                         ^ source.swarm expression.group
#                          ^ source.swarm expression.group variable.parameter
#                           ^ source.swarm expression.group punctuation.paren.close
#                            ^ source.swarm
#                             ^^^ source.swarm keyword.control.dictionary.def.end
>
>lit 15;
#^^^ source.swarm variable.parameter
#   ^ source.swarm
#    ^^ source.swarm constant.numeric
#      ^ source.swarm keyword.operator.chain
>
>lit 0xf;
#^^^ source.swarm variable.parameter
#   ^ source.swarm
#    ^ source.swarm constant.numeric
#     ^^ source.swarm
#       ^ source.swarm keyword.operator.chain
>
>lit 0b1111;
#^^^ source.swarm variable.parameter
#   ^ source.swarm
#    ^ source.swarm constant.numeric
#     ^ source.swarm
#      ^^^^ source.swarm constant.numeric
#          ^ source.swarm keyword.operator.chain
>
>lit " Hello ";
#^^^ source.swarm variable.parameter
#   ^ source.swarm
#    ^ source.swarm string.quoted.double
#     ^^^^^^^ source.swarm string.quoted.double
#            ^ source.swarm string.quoted.double
#             ^ source.swarm keyword.operator.chain
>
>lit "Hello\nWorld!";
#^^^ source.swarm variable.parameter
#   ^ source.swarm
#    ^ source.swarm string.quoted.double
#     ^^^^^ source.swarm string.quoted.double
#          ^^ source.swarm string.quoted.double constant.character.escape.sequence.swarm
#            ^^^^^^ source.swarm string.quoted.double
#                  ^ source.swarm string.quoted.double
#                   ^ source.swarm keyword.operator.chain
>